### DISCUSSION TITLE
exposing uniqueIdent and toAsciiText to lua api in filters

### DISCUSSION THREAD

**Discussion by massifrg** [jgm/pandoc#10353] [Ideas]
This is a feature request.

`uniqueIdent` and `toAsciiText` are functions used internally by pandoc to generate identifiers, but only for formats that support the `auto_identifiers` and `ascii_identifiers` extensions.

They could be useful in lua filters too.

In my particular case, I'm looking for a function that takes a list of Inlines and transforms it to a normalized (low case, ascii chars only) string. I'm going to use it to automatically generate a [sort-key](https://github.com/massifrg/pandoc-export-index#terms-topics) attribute for index terms, to be manually refined by the user to tune the sorting of an index.

They could go under `pandoc.utils` or `pandoc.text` packages.

**Comment by tarleb** [comment]
Exposing `uniqueIdent` makes perfect sense, feel free to open an issue for that. In case you also want to provide a PR: the necessary code would go into `pandoc-lua-engine/src/Text/Pandoc/Lua/Module/Utils.hs`. It should be just a couple of lines.

I'm less certain about `toAsciiText`, but I guess adding that to `pandoc.text` is reasonable.

**Reply by massifrg** [reply]
Thank you @tarleb for your guidance, I'm going to make a PR.

Just yesterday I recompiled pandoc from source for the first time.
Months ago, I already tried to read and understand the code under `pandoc-lua-engine`.
Though I understood the logic, I was not able to add any code.
This task seems to be feasible; let's see...

**Reply by tarleb** [reply]
The `make_sections` function is probably closest to what's needed. Feel free to ping me directly if you run into problems.
